samples/charts/pie-chart/animation/Pages/PieChartAnimation.razor

```diff
@@ -0,0 +1,101 @@
+﻿@page "/PieChartAnimation"
+@* this namespace is required for Infragistics controls *@
+@using Infragistics.Blazor.Controls
+@inject IInfragisticsBlazor InfragisticsBlazor;
+@inject IJSRuntime JSRuntime;
+
+<div class="igContainer">
+
+    <div class="igOptions-horizontal">
+        <button @onclick="onAnimationToggle">Animate Chart</button>
+    </div>
+
+    <div class="igComponent">
+
+        @if (DataSource != null)
+        {
+            <PieChart Height="600px" Width="600px"
+                      DataSource="DataSource"
+                      LabelMemberPath="Summary"
+                      ValueMemberPath="MarketShare"
+                      LegendLabelMemberPath="Company"
+                      LabelsPosition="LabelsPosition.InsideEnd"
+                      StartAngle="StartAngle"
+                      LabelExtent="LabelExtent"
+                      RadiusFactor="RadiusFactor">
+            </PieChart>
+        }
+
+    </div>
+</div>
+
+@code {
+
+    private List<DataItem> DataSource;
+    public double StartAngle = 0;
+    public double RadiusFactor = 0.1;
+    public double LabelExtent = 0.7;
+    public Boolean isAnimating = false;
+
+    public void onAnimationToggle()
+    {
+        if (!this.isAnimating)
+        {
+            this.StartAngle = 0;
+            this.RadiusFactor = 0.1;
+            this.isAnimating = true;
+
+            Task.Delay(16).ContinueWith((t) => OnTimerTick());
+        }
+        else
+        {
+            this.isAnimating = false;
+        }
+    }
+
+    protected override void OnInitialized()
+    {
+        base.OnInitialized();
+        PieChartModule.Register(InfragisticsBlazor);
+
+        var dataItems = new List<DataItem>();
+        dataItems.Add(new DataItem() { MarketShare = 30, Company = "Google", });
+        dataItems.Add(new DataItem() { MarketShare = 25, Company = "Apple", });
+        dataItems.Add(new DataItem() { MarketShare = 20, Company = "Microsoft", });
+        dataItems.Add(new DataItem() { MarketShare = 15, Company = "Samsung", });
+        dataItems.Add(new DataItem() { MarketShare = 10, Company = "Other", });
+
+        DataSource = dataItems;
+        this.onAnimationToggle();
+    }
+
+    private void OnTimerTick()
+    {
+        if (this.isAnimating)
+        {
+            if (this.RadiusFactor < 1.0)
+                this.RadiusFactor += 0.0025;
+
+            if (this.StartAngle < 360)
+                this.StartAngle++;
+
+            if (this.RadiusFactor >= 1.0 &&
+                this.StartAngle >= 360)
+            {
+                this.isAnimating = false;
+            }
+
+            Task.Delay(16).ContinueWith((t) => OnTimerTick());
+        }
+
+        this.StateHasChanged();
```

Comment:
did you verify that OnTimerTick is not called when the sample completes animation? 
For example, adding here: Console.WriteLine("OnTimerTick  " + this.StartAngle);
and checking if this log message stops at 360

Reply:
move `this.StateHasChanged();` after `Task.Delay(16).ContinueWith((t) => OnTimerTick());`